tests/test_types.py

```diff
@@ -0,0 +1,81 @@
+from dataclasses import dataclass
+from sys import version_info
+from typing import Optional, Union
+
+from pytest import mark, raises
+
+from pydargs import parse
+
+
+class TestUnion:
+    @dataclass
+    class Config:
+        a: int
+        b: Optional[int] = None
+        c: Union[int, str] = "b"
+        d: Union[None, int] = None
+        e: Optional[Union[int, float]] = None
+        f: float = 2.0
+
+    def test_default(self):
+        config = parse(self.Config, ["--a", "1"])
+        assert config.a == 1
+        assert config.b is None
+        assert config.c == "b"
+        assert config.d is None
+        assert config.e is None
+
+    def test_parse_optional(self):
+        config = parse(self.Config, ["--a", "1", "--b", "2"])
+        assert config.b == 2
+
+    def test_parse_optional_invalid(self, capsys):
+        with raises(SystemExit):
+            parse(self.Config, ["--a", "1", "--b", "2.0"])
+        captured = capsys.readouterr()
+        assert "argument --b: invalid functools.partial" in captured.err
```

Comment:
Is there anything we can do about this error? It's not very informative 😢

Reply:
How about now?